core/client/src/main/java/alluxio/client/block/BlockWorkerClient.java

```diff
@@ -378,11 +379,7 @@ public Boolean call() throws AlluxioTException, TException {
         }
       });
     } catch (AlluxioException e) {
-      if (e instanceof WorkerOutOfSpaceException) {
-        return false;
-      } else {
-        throw new IOException(e);
-      }
+      throw new IOException(e);
```

Comment:
Does the `@return` javadoc already cover this issue? It says
```java
   * @return true if space was successfully allocated, false if the worker is unable to allocate
   *         space due to space exhaustion
```
so I think it should be pretty clear that an exception will not be thrown if the worker runs out of space. My concern with adding the comment is that it's superfluous given the javadoc and could easily be overlooked even if the worker-side contract does change.